.github/workflows/docker.yml

```diff
@@ -0,0 +1,88 @@
+name: Docker
+
+on:
+  workflow_call:
+    secrets:
+      DOCKERHUB_TOKEN:
+        required: true
+      DOCKERHUB_USERNAME:
+        required: true
+
+jobs:
+  docker-meta:
+    runs-on: ubuntu-latest
+    outputs:
+      labels: ${{ steps.meta-main.outputs.labels || steps.meta-version.outputs.labels }}
+      tags: ${{ steps.meta-main.outputs.tags || steps.meta-version.outputs.tags }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Docker meta
+        id: meta-main
+        uses: docker/metadata-action@v4
+        if: startsWith(github.ref, 'refs/tags/v') || (github.ref == 'refs/heads/main')
+        with:
+          images: |
+            solidproject/community-server
+          # edge will always be executed (without latest tag), semver only on tag push events (with latest tag)
+          tags: |
+            type=edge
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+          github-token: ${{ secrets.github_token }}
+      - name: Docker meta
+        id: meta-version
+        uses: docker/metadata-action@v4
+        if: startsWith(github.ref, 'refs/heads/versions/')
+        with:
+          images: |
+            solidproject/community-server
+          # Just one label: next (no latest here) for the last pushed commit on this branch
+          tags: |
+            type=raw,value=next
+          github-token: ${{ secrets.github_token }}
+
+  docker:
+    needs: docker-meta
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and export to docker
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          load: true
+          tags: ${{ needs.docker-meta.outputs.tags }}
+          labels: ${{ needs.docker-meta.outputs.labels }}
```

Comment:
Not sure but should this be `needs` and not `jobs.docker-meta.outputs.tags` ?

Same for further mentions of `needs.`. I am probably wrong, but never heard of this syntax before.

Reply:
Yea it can be confusing when going through their docs but the `jobs.<job_id>.outputs` path is for defining the outputs, using them in other jobs uses the `needs.<job_id>.outputs` path as you need to depend on the job to get the output

See https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs#overview